File: TimeUtil.java
import java.time.LocalDateTime;
import java.time.ZoneId;
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;
import java.util.regex.Pattern;

public class TimeUtil {
    // 固定统一格式
    private static final String PATTERN = "yyyy-MM-dd-HH-mm-ss";
    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern(PATTERN);

    /**
     * 获取当前标准时间
     */
    public static String now() {
        return LocalDateTime.now(ZoneId.systemDefault()).format(FORMATTER);
    }

    /**
     * 万能格式化，兼容各种奇葩时间、时间戳
     */
    public static String format(Object timeObj) {
        if (timeObj == null) {
            return now();
        }
        String time = timeObj.toString();

        // 处理时间戳
        if (Pattern.matches("\\d+", time)) {
            long ts = Long.parseLong(time);
            LocalDateTime dt;
            if (time.length() == 10) {
                dt = LocalDateTime.ofEpochSecond(ts, 0, ZoneId.systemDefault().getRules().getOffset(LocalDateTime.now()));
            } else {
                dt = LocalDateTime.ofEpochSecond(ts / 1000, 0, ZoneId.systemDefault().getRules().getOffset(LocalDateTime.now()));
            }
            return dt.format(FORMATTER);
        }

        // 统一清洗分隔符
        time = time.replace("/", "-")
                   .replace(".", "-")
                   .replace("T", "-")
                   .replace(":", "-");

        try {
            LocalDateTime dateTime = LocalDateTime.parse(time, DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss"));
            return dateTime.format(FORMATTER);
        } catch (DateTimeParseException e) {
            return now();
        }
    }
}

File: TimeUtil.py
import re
from datetime import datetime

class TimeUtil:
    # 全局统一格式
    FORMAT = "%Y-%m-%d-%H-%M-%S"

    @staticmethod
    def now() -> str:
        """获取当前标准时间"""
        return datetime.now().strftime(TimeUtil.FORMAT)

    @staticmethod
    def format(time_input=None) -> str:
        """
        万能时间格式化
        支持：时间戳(秒/毫秒)、2025/xx/xx、2025.xx.xx、带T、中文日期
        """
        if time_input is None:
            return TimeUtil.now()

        try:
            # 处理数字时间戳
            if isinstance(time_input, (int, float)):
                ts = int(time_input)
                # 10位秒 / 13位毫秒
                if len(str(ts)) == 10:
                    dt = datetime.fromtimestamp(ts)
                else:
                    dt = datetime.fromtimestamp(ts / 1000)
                return dt.strftime(TimeUtil.FORMAT)

            # 统一替换分隔符 兼容 / . 、
            text = str(time_input)
            text = re.sub(r"[/\.年月]", "-", text)
            text = re.sub(r"[T时:分]", "-", text)

            dt = datetime.fromisoformat(text)
            return dt.strftime(TimeUtil.FORMAT)
        except Exception:
            return TimeUtil.now()
